### PR TITLE
Make sure when reading int data, it gets properly converted to float

### PR DESCRIPTION
**Pull request by JonathanMeans**

### CODE REVIEW

File: six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+        const std::vector<double>& scaleFactors,
+        bool scale,
+        const types::RowCol<size_t>& dims)
+{
+    cphd::CPHDReader reader(pathname, numThreads);
+    cphd::Wideband& wideband = reader.getWideband();
+    std::vector<std::complex<float> > readData(dims.area());
+
+    size_t sizeInBytes = readData.size() * sizeof(readData[0]);
+    sys::ubyte scratchData[sizeInBytes];
```
Comment (asylvest) [inline six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp]: Can't create an array on the stack whose size you don't know at compile time (gcc is being nice).  Use a vector or ScopedArray.

File: six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+{
+    try
+    {
+        TEST_CHECK(testUnscaledInt8);
+        TEST_CHECK(testScaledInt8);
+        TEST_CHECK(testUnscaledInt16);
+        TEST_CHECK(testScaledInt16);
+        TEST_CHECK(testUnscaledFloat);
+        TEST_CHECK(testScaledFloat);
+        TempFile tempfile;
```
Comment (asylvest) [inline six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp]: Why is this `TempFile` here?

Comment (asylvest) [inline six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp]: Program should return 0 on success

File: six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+#include <cphd/CPHDReader.h>
+#include <cphd/CPHDWriter.h>
+#include <types/RowCol.h>
+#include <cli/ArgumentParser.h>
+
+#include "TestCase.h"
+
+namespace
+{
+class TempFile
```
Comment (asylvest) [inline six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp]: Let's wait and get the TempFile support from CODA-OSS merged in and then we can wipe this

File: six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp (hunk 2 above)
Comment (asylvest) [inline six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp]: Can you update the Python test script to call this application?  Or is something else running the unittests automatically right now?

Comment (JonathanMeans) [inline six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp]: The unittests get run automatically.

**Comment by JonathanMeans** [comment]
This is ready

**Comment by asylvest** [comment]
Looks good

**Pull request merged**